mayastor/src/bin/casperf.rs

```diff
@@ -0,0 +1,405 @@
+            total_mb_per_second += mb_per_second;
+        }
+
+        println!("\r ==================================================== +");
+        println!(
+            "\r {:20}: {:10} IO/s {:10}: MB/s\n",
+            "Total", total_io_per_second, total_mb_per_second
+        );
+    });
+    0
+}
+
+fn main() {
+    logger::init("INFO");
+
+    // dont not start the target(s)
+    Config::get_or_init(|| {
+        let mut cfg = Config::default();
+        cfg.nexus_opts.iscsi_enable = false;
+        cfg.nexus_opts.nvmf_enable = false;
+        cfg.sync_disable = true;
+        cfg
+    });
+
+    let matches = App::new("\nMayastor performance tool")
+        .version("0.1")
+        .settings(&[AppSettings::ColoredHelp, AppSettings::ColorAlways])
+        .about("Perform IO to storage URIs")
+        .arg(
+            Arg::with_name("io_size")
+                .value_name("io_size")
+                .short("b")
+                .help("block size in bytes")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("queue_depth")
+                .value_name("queue_depth")
+                .short("q")
+                .help("queue depth")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("URI")
+                .value_name("URI")
+                .help("storage URI's")
+                .index(1)
+                .multiple(true)
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let mut uris = matches
+        .values_of("URI")
+        .unwrap()
+        .map(|u| u.to_string())
+        .collect::<Vec<_>>();
+
+    let io_size = value_t!(matches.value_of("io_size"), u64).unwrap_or(IO_SIZE);
+    let qd = value_t!(matches.value_of("queue_depth"), u64).unwrap_or(QD);
+    let mut args = MayastorCliArgs::default();
+
+    args.reactor_mask = "0x2".to_string();
+    //args.grpc_endpoint = Some("0.0.0.0".to_string());
+
+    let ms = MayastorEnvironment::new(args);
+    ms.start(move || {
+        sig_override();
+
+        Reactors::master().send_future(async move {
+            let jobs = uris
+                .iter_mut()
+                .map(|u| Job::new(u, io_size, qd))
+                .collect::<Vec<_>>();
+
+            for j in jobs {
+                let job = j.await;
+                let thread =
+                    Mthread::new(job.bdev.name(), Cores::current()).unwrap();
+                thread.msg(job, |job| {
+                    job.run();
+                });
+            }
+
+            unsafe {
+                PERF_TICK.with(|p| {
+                    *p.borrow_mut() =
+                        NonNull::new(spdk_sys::spdk_poller_register(
+                            Some(perf_tick),
+                            std::ptr::null_mut(),
+                            1_000_000,
+                        ))
+                });
+            }
+        });
+
+        // kick of a task for each core if wanted here
```

Comment:
typo - "kick **off**"

Reply:
i removed the whole comment as it was a leftover thanks :+1: